msnoise/preprocessing.py

```diff
@@ -15,10 +15,45 @@
 
 from .api import *
 
+# Function returns a pz dict for the total response from the inventory response object
+def get_pazdictfrominventory(inv):
+    polezerostage=inv.get_paz()
+    totalsensitivity=inv.instrument_sensitivity
+    pzdict={}
+    pzdict['poles']=polezerostage.poles
+    pzdict['zeros']=polezerostage.zeros
+    pzdict['gain']=polezerostage.normalization_factor
+    pzdict['sensitivity']=totalsensitivity.value
+    return pzdict
 
 def preprocess(db, stations, comps, goal_day, params):
     datafiles = {}
     output = Stream()
+
+    # Preload the response information
+    if get_config(db, 'remove_response', isbool=True):
+        logging.debug('Loading response information for job %s' % goal_day)
+        response_format = get_config(db, 'response_format')
+        response_prefilt = eval(get_config(db, 'response_prefilt'))
+        files = glob.glob(os.path.join(get_config(db,'response_path'),"*"))
+        if response_format == "inventory":
+            firstinv = True
+            inventory = None
+            for file in files:
+                try:
+                    inv = read_inventory(file)
+                    if firstinv:
+                        inventory = inv
+                        firstinv = False
+                    else:
+                        inventory += inv
+                except:
+                    traceback.print_exc()
+                    pass
+        elif response_format == "dataless":
+            for file in files:
+                p = Parser(file)
```

Comment:
it will fail here if there are more than 1 dataless file. It'll load the "last one" in `p` and then, later in the calls, the get_paz won't succeed.